app/Jobs/Invoice/InvoiceWorkflowSettings.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * Invoice Ninja (https://invoiceninja.com)
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2020. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://opensource.org/licenses/AAL
+ */
+
+namespace App\Jobs\Invoice;
+
+use App\Mail\Invoices\InvoiceWasPaid;
+use App\Models\Client;
+use App\Models\Invoice;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Mail;
+
+class InvoiceWorkflowSettings implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public $invoice;
+    public $client;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Invoice $invoice, Client $client = null)
+    {
+        $this->invoice = $invoice;
+        $this->client = $client ?? $invoice->client;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        if ($this->client->getSetting('auto_archive_invoice')) {
+            $this->invoice->archive();
+        }
+
+        if ($this->client->getSetting('auto_email_invoice')) {
+           // Todo: Fetch the right client contact.
+            Mail::to($this->client->contacts()->first()->email)
```

Comment:
I think we should use

$invoice->service()->sendEmail() here as this will iterate over all the contacts that need to be sent this invoice.

Reply:
sendEmail() looks nice, however it accepts ClientContact instead of Client. I don't see any loop there as well.

![image](https://user-images.githubusercontent.com/13711415/80369622-de07ad00-888e-11ea-9c5b-afd121df136d.png)

Reply:
Ah yes, we would need to iterate over the invoice->invitations->each() and then fire the sendEmail($contact) method as we have access to the contact inside the invitation.